hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2548,3 +2548,25 @@ func (r *NodePoolReconciler) getPullSecretName(ctx context.Context, hostedCluste
 	}
 	return pullSecret.Name, nil
 }
+
+// machinesByCreationTimestamp sorts a list of Machine by creation timestamp, using their names as a tie breaker.
+type machinesByCreationTimestamp []*capiv1.Machine
+
+func (o machinesByCreationTimestamp) Len() int      { return len(o) }
+func (o machinesByCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o machinesByCreationTimestamp) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
+		return o[i].Name < o[j].Name
+	}
+	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
+}
+
+// SortedByCreationTimestamp returns the machines sorted by creation timestamp.
+func sortedByCreationTimestamp(machines []*capiv1.Machine) []*capiv1.Machine {
+	res := make(machinesByCreationTimestamp, 0, len(machines))
```

Comment:
this function can be simplified:
```
sort.Sort(machinesByCreationTimestamp(machines))
return machines
```
or is there a reason you want to keep the original slice unmodified?